lib/doorkeeper/orm/active_record/application.rb

```diff
@@ -11,14 +11,9 @@ class Application < ActiveRecord::Base
     end
     has_many :authorized_applications, through: :authorized_tokens, source: :application
 
-    def self.column_names_with_table
-      self.column_names.map { |c| "#{table_name}.#{c}" }
-    end
-
     def self.authorized_for(resource_owner)
-      joins(:authorized_applications).
-        where(AccessToken.table_name => { resource_owner_id: resource_owner.id, revoked_at: nil }).
-        group(column_names_with_table.join(','))
+      where(id: Doorkeeper::AccessToken.
+        where(resource_owner_id: resource_owner.id, revoked_at: nil))
     end
```

Comment:
But we don't want to match `oauth_applications.id` with `oauth_access_tokens.id`, we want to match them with `oauth_access_tokens.application_id` instead. I'm surprised the tests didn't fail.

> I don't think there is a need in extracting of the `Doorkeeper::AccessToken.where`, but if you want - I can do it.

I'm trying to find a blog post that explains how writing one statement per line makes code easier to reason about. Will post when I find it.